packages/govern-server/src/core/utils/network.ts

```diff
@@ -1,84 +1,102 @@
+    ensAddress?: Address
+    name?: string
+  }) {
+    if (name === undefined && chainId === undefined) {
+      throw new ErrorInvalidNetwork(
+        `Network: no name or chainId passed. ` +
+          `Please provide at least one of these.`
+      )
+    }
 
-function networkFromObject({
-  chainId,
-  ensAddress,
-  name,
-}: {
-  chainId?: number
-  ensAddress?: Address
-  name?: string
-}): Network {
-  if (name === undefined && chainId === undefined) {
-    throw new ErrorInvalidNetwork(
-      `Network: no name or chainId passed. ` +
-        `Please provide at least one of these.`
-    )
-  }
+    // Handle the case of having a name but no chainId.
+    if (name !== undefined && chainId === undefined) {
+      this.#chainId = this.networkFromName(name)?.chainId
+
+      if (chainId === undefined) {
+        throw new ErrorInvalidNetwork(
+          `Network: invalid name provided: ${name}. ` +
+            `Please use provide a chainId or use one of the following names: ` +
+            config.networks.map((network) => network.chainId).join(', ') +
+            `.`
+        )
+      }
+    }
 
-  // Handle the case of having a name but no chainId.
-  if (name !== undefined && chainId === undefined) {
-    chainId = networkFromName(name)?.chainId
+    // Just a little help for TypeScript, at this
+    // point we know that chainId cannot be undefined.
+    this.#chainId = chainId as number
 
-    if (chainId === undefined) {
+    const chainIdNetwork = this.networkFromChainId(chainId)
+
+    if (!chainIdNetwork) {
       throw new ErrorInvalidNetwork(
-        `Network: invalid name provided: ${name}. ` +
-          `Please use provide a chainId or use one of the following names: ` +
-          NETWORKS.map((network) => network.chainId).join(', ') +
+        `Network: invalid chainId provided: ${chainId}. ` +
+          `Please use one of the following: ` +
+          config.networks.map((network) => network.chainId).join(', ') +
           `.`
       )
     }
-  }
 
-  // Just a little help for TypeScript, at this
-  // point we know that chainId cannot be undefined.
-  chainId = chainId as number
+    this.#name = name ?? chainIdNetwork.name
+    this.#ensAddress = ensAddress ?? chainIdNetwork.ensAddress
+  }
 
-  const chainIdNetwork = networkFromChainId(chainId)
+  get chainId() {
+    return this.#chainId
+  }
 
-  if (!chainIdNetwork) {
-    throw new ErrorInvalidNetwork(
-      `Network: invalid chainId provided: ${chainId}. ` +
-        `Please use one of the following: ` +
-        NETWORKS.map((network) => network.chainId).join(', ') +
-        `.`
-    )
+  get ensAddress() {
+    return this.#ensAddress
   }
 
-  // We compare with undefined to accept empty strings.
-  if (name === undefined) {
-    name = chainIdNetwork.name
+  get name() {
+    return this.#name
   }
 
-  if (!ensAddress) {
-    ensAddress = chainIdNetwork.ensAddress
+  private networkFromChainId(chainId: number): NetworkType | null {
```

Comment:
Should `networkFromChainId` and `networkFromName` be simple functions or static methods, since they are not really related to `Network` instances?